Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -363,8 +365,49 @@
     m_scrollContainerLayer->setEventRegion(WTFMove(eventRegion));
 }
 
-bool UnifiedPDFPlugin::handleMouseEvent(const WebMouseEvent&)
+enum class AltKeyIsActive : bool { No, Yes };
+
+static WebCore::Cursor::Type toWebCoreCursorType(UnifiedPDFPlugin::ElementsUnderMouse elementsUnderMouse, AltKeyIsActive altKeyIsActive = AltKeyIsActive::No)
+{
+    auto cursor = WebCore::Cursor::Type::Pointer;
+
+    if (elementsUnderMouse.containsAny({ UnifiedPDFPlugin::ElementUnderMouse::Link, UnifiedPDFPlugin::ElementUnderMouse::Control, UnifiedPDFPlugin::ElementUnderMouse::Icon }))
+        cursor = WebCore::Cursor::Type::Hand;
+    else if (altKeyIsActive == AltKeyIsActive::Yes)
+        cursor = WebCore::Cursor::Type::Hand;
+    else if (elementsUnderMouse.contains(UnifiedPDFPlugin::ElementUnderMouse::Text))
+        cursor = WebCore::Cursor::Type::IBeam;
+    else if (elementsUnderMouse.contains(UnifiedPDFPlugin::ElementUnderMouse::TextField))
+        cursor = WebCore::Cursor::Type::IBeam;
+
+    return cursor;
+}
+
+UnifiedPDFPlugin::ElementsUnderMouse UnifiedPDFPlugin::getElementsUnderMouse(const WebMouseEvent&) const
 {
+    return { };
+}
+
+bool UnifiedPDFPlugin::handleMouseEvent(const WebMouseEvent& event)
+{
+    switch (event.type()) {
+    case WebEventType::MouseMove:
+        mouseMovedInContentArea();
+        switch (event.button()) {
+        case WebMouseEventButton::None:
+        {
+            auto altKeyIsActive = event.altKey() ? AltKeyIsActive::Yes : AltKeyIsActive::No;
+            auto elementsUnderMouse = getElementsUnderMouse(event);
+            notifyCursorChanged(toWebCoreCursorType(elementsUnderMouse, altKeyIsActive));
+            return true;
+        }
+        default:
+            return false;
+        }
+    default:
```

Comment:
this default can just be return false and then the bottom-most return false can be omitted

Reply:
Fixed